pkg/sink/sink_test.go

```diff
@@ -478,3 +481,59 @@ func Test_addLabels(t *testing.T) {
 		t.Error(diff)
 	}
 }
+
+type fakeHTTPInterceptor struct {
+	i int32
+}
+
+func (f *fakeHTTPInterceptor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	data := map[string]interface{}{
+		"i": atomic.AddInt32(&f.i, 1),
+	}
+	_ = json.NewEncoder(w).Encode(data)
+}
+
+func TestExecuteInterceptor(t *testing.T) {
```

Comment:
might be useful to include a brief docstring explaining how this test worked - after reading through it it seems super cool ✨ but had to read a bunch before understanding it (that the interceptor increments an int etc)

Reply:
Done.